compilers/concrete-compiler/compiler/lib/Common/Transformers.cpp

```diff
@@ -195,30 +195,45 @@ updateGateInfoAccordingValue(Message<concreteprotocol::GateInfo> &gate,
   }
   auto gateCiphertext = gateTypeInfo.getLweCiphertext();
   auto gateCompression = gateCiphertext.getCompression();
-  if (gateCompression == concreteprotocol::Compression::SEED) {
-    auto valueReader = value.asReader();
-    auto valueTypeInfo = valueReader.getTypeInfo();
-    if (!valueTypeInfo.hasLweCiphertext()) {
-      return gate;
-    }
-    auto valueCiphertext = valueTypeInfo.getLweCiphertext();
-    auto valueCompression = valueCiphertext.getCompression();
-    if (valueCompression == concreteprotocol::Compression::NONE) {
-      // If the compression of transportValue is none and the gateInfo have
-      // compression we update the gateInfo to allow uncompressed transportValue
-      auto gateBuilder = gate.asBuilder();
-      gateBuilder.getTypeInfo().getLweCiphertext().setCompression(
-          concreteprotocol::Compression::NONE);
-      auto gateDimensions = gateBuilder.getRawInfo().getShape().getDimensions();
-      auto lweSize = gateCiphertext.getEncryption().getLweDimension() + 1;
-      gateDimensions.set(gateDimensions.size() - 1, lweSize);
-      auto concreteShapeDimensions = gateBuilder.getTypeInfo()
-                                         .getLweCiphertext()
-                                         .getConcreteShape()
-                                         .getDimensions();
-      concreteShapeDimensions.set(concreteShapeDimensions.size() - 1, lweSize);
-      return gateBuilder.asReader();
-    }
+  auto valueReader = value.asReader();
+  auto valueTypeInfo = valueReader.getTypeInfo();
+  if (!valueTypeInfo.hasLweCiphertext()) {
+    return gate;
+  }
+  auto valueCiphertext = valueTypeInfo.getLweCiphertext();
+  auto valueCompression = valueCiphertext.getCompression();
+  if (gateCompression == concreteprotocol::Compression::SEED &&
+      valueCompression == concreteprotocol::Compression::NONE) {
```

Comment:
most of the line in the if could be moved outside of the if block. they are shared between the 2 cases.
Or the 2 case could be unified sice they only differs on 
lweSize vs 3
setCompression(valueCompression)